### PR TITLE
ceph: Added some extra 'nil' checks

### PR DESCRIPTION
**Pull request by aruniiird**
Signed-off-by: Arun Kumar Mohan <amohan@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/cluster/cluster_external.go [hunk 1]
```diff
@@ -45,6 +45,11 @@ func (c *ClusterController) configureExternalCephCluster(cluster *cluster) error
 	// loop until we find the secret necessary to connect to the external cluster
 	// then populate clusterInfo
 	cluster.Info = mon.PopulateExternalClusterInfo(c.context, c.namespacedName.Namespace)
+	if cluster.Info == nil {
```
Comment (travisn) [inline pkg/operator/ceph/cluster/cluster_external.go:48]: Will clusterInfo ever be nil here? I thought `PopulateExternalClusterInfo()` didn't return until it was available.

File: pkg/operator/ceph/cluster/cluster_external.go [hunk 2]
```diff
@@ -45,6 +45,11 @@ func (c *ClusterController) configureExternalCephCluster(cluster *cluster) error
...
+		err := errors.New("Unable to retrieve a valid cluster info")
+		logger.Errorf("Error: %+v", err)
```
Comment (travisn) [inline pkg/operator/ceph/cluster/cluster_external.go:50]: The caller should log the error so we shouldn't need to log it here.

File: pkg/operator/ceph/cluster/cluster_external.go [hunk 3]
```diff
@@ -173,6 +178,9 @@ func purgeExternalCluster(clientset kubernetes.Interface, namespace string) {
 }
 
 func validateExternalClusterSpec(cluster *cluster) error {
+	if cluster == nil {
```
Comment (travisn) [inline pkg/operator/ceph/cluster/cluster_external.go:181]: I don't think `cluster` will ever be nil. Did you mean to check for `cluster.Info`?
```suggestion
	if cluster.Info == nil {
```

**Comment by aruniiird** [comment]
I added those checks according to some extra logs which I added to trace an issue and `defer logger.Info()` logged `falsely` a nil representation of the `clusterInfo`. That was a false alarm and this fix was more or less trying fix a non-existing scenario.
So closing the PR.

**Pull request closed without merging**